docs/docs/spec.md

```diff
@@ -1038,6 +1038,21 @@ The Java rules are defined in [JLS 5.1.10]. We add to them as follows:
 
     > See ["Augmented null types."](#null-types)
 
+## Harmonizing annotations on special type usages {#harmoize-annotations}
+
+> Because of a special case [above](#augmented-type-of-usage) that makes record
+> `equals` parameters nullable, there is a mismatch between what `public boolean
+> equals(Object o)` means in a record class and what it would otherwise mean. To
+> harmonize this, we anticipate that tools will sometimes still produce errors
+> unless authors include the usual `@Nullable` annotation on the parameter.
```

Comment:
Maybe there's a clearer thing to say than "there is a mismatch between what X means in a record class and what it would otherwise mean."

What about something like this:

```suggestion
> Because of the special case [above](#augmented-type-of-usage) that makes record
> `equals` parameters always nullable, we include this rule so that tools can produce
> expected errors in some cases when the parameter is not annotated with `@Nullable`.
```